desktop/ui/src/main/java/org/datacleaner/user/DataCleanerHomeUpgrader.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.user;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.commons.vfs2.AllFileSelector;
+import org.apache.commons.vfs2.FileDepthSelector;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileType;
+import org.apache.metamodel.util.FileHelper;
+import org.datacleaner.Version;
+import org.datacleaner.VersionComparator;
+import org.datacleaner.util.ResourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Checks if there is DataCleanerHome set up for previous versions of
+ * DataCleaner and upgrades.
+ */
+public final class DataCleanerHomeUpgrader {
+    
+    private static final Logger logger = LoggerFactory.getLogger(DataCleanerHomeUpgrader.class);
+
+    private static final List<String> candidateBlacklist = Arrays.asList("log", Version.UNKNOWN_VERSION);
+
+    public boolean upgrade(FileObject target) {
```

Comment:
Would be nice with a description of what (folder I assume) it is. I am not sure if it is an old folder, the new candidate or maybe their common ancestor?